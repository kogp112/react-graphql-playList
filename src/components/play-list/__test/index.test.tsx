import * as React from "react";
import * as renderer from "react-test-renderer";
import PlayList from "../index";

let container: JSXElement = null;
type JSXElement = Element | null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container = null;
});

it("should render play list", () => {
  const onChange = jest.fn();
  const list = {
    name: "90sPunk",
    songs: [
      {
        imageUrl: "https://img.youtube.com/vi/fWtFpYcgRFs/0.jpg",
        name: "Tilt - Fool To Blame",
        url: "https://www.youtube.com/watch?v=fWtFpYcgRFs",
      },
      {
        imageUrl: "https://img.youtube.com/vi/fWtFpYcgRFs/0.jpg",
        name: "Tilt - Fool To Blame",
        url: "https://www.youtube.com/watch?v=fWtFpYcgRFs",
      },
      {
        imageUrl: "https://img.youtube.com/vi/fWtFpYcgRFs/0.jpg",
        name: "Tilt - Fool To Blame",
        url: "https://www.youtube.com/watch?v=fWtFpYcgRFs",
      },
    ],
  };

  const component = renderer
  .create(<PlayList list={list} handleClickPlayList={onChange} />)
  .toJSON();
  expect(component).toMatchSnapshot();
});
