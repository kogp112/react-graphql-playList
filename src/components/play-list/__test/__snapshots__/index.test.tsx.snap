// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render play list 1`] = `
<div
  className="MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      90sPunk
    </h5>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-134 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        />
        <div
          className="MuiCardHeader-action"
        >
          <button
            aria-label="play/pause"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root makeStyles-playIcon-137"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiCardContent-root makeStyles-content-135"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
        >
          Tilt - Fool To Blame
        </h6>
      </div>
      <img
        className="MuiCardMedia-root makeStyles-cover-136 MuiCardMedia-media MuiCardMedia-img"
        src="https://img.youtube.com/vi/fWtFpYcgRFs/0.jpg"
        title="Tilt - Fool To Blame"
      />
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-198 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        />
        <div
          className="MuiCardHeader-action"
        >
          <button
            aria-label="play/pause"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root makeStyles-playIcon-201"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiCardContent-root makeStyles-content-199"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
        >
          Tilt - Fool To Blame
        </h6>
      </div>
      <img
        className="MuiCardMedia-root makeStyles-cover-200 MuiCardMedia-media MuiCardMedia-img"
        src="https://img.youtube.com/vi/fWtFpYcgRFs/0.jpg"
        title="Tilt - Fool To Blame"
      />
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-202 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        />
        <div
          className="MuiCardHeader-action"
        >
          <button
            aria-label="play/pause"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root makeStyles-playIcon-205"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiCardContent-root makeStyles-content-203"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
        >
          Tilt - Fool To Blame
        </h6>
      </div>
      <img
        className="MuiCardMedia-root makeStyles-cover-204 MuiCardMedia-media MuiCardMedia-img"
        src="https://img.youtube.com/vi/fWtFpYcgRFs/0.jpg"
        title="Tilt - Fool To Blame"
      />
    </div>
  </div>
</div>
`;
